windows/RNPrint/RNPrint.h

```diff
@@ -0,0 +1,402 @@
+                        }
+                      );
+                      printDoc.AddPagesComplete();
+                    }
+                  );
+                }
+              );
+
+            });
+
+          printToken = printMan.PrintTaskRequested(
+            [=](PrintManager const& sender, PrintTaskRequestedEventArgs const& args)
+            {
+              auto printTask = args.Request().CreatePrintTask(winrt::to_hstring(capturedOptions.jobName),
+                [=](PrintTaskSourceRequestedArgs const& args)
+                {
+                  args.SetSource(printDocSource);
+                }
+              );
+              if (options.isLandscape)
+              {
+                printTask.Options().Orientation(PrintOrientation::Landscape);
+              } else
+              {
+                printTask.Options().Orientation(PrintOrientation::Portrait);
+              }
+
+              printTask.Completed([=](PrintTask const& sender, PrintTaskCompletedEventArgs const& args)
+                {
+                  cleanUp();
+                  switch (args.Completion())
+                  {
+                  case PrintTaskCompletion::Failed:
+                    capturedPromise.Reject("Failed to print.");
+                    break;
+                  case PrintTaskCompletion::Abandoned:
+                    capturedPromise.Reject("Printing Abandoned");
+                    break;
+                  default:
+                    capturedPromise.Resolve(options.jobName);
+                    break;
+                  }
+                }
+              );
+            }
+          );
+
+          co_await PrintManager::ShowPrintUIAsync();
+        }
+      }
+    }
+
+    REACT_METHOD(Print, L"print")
+      void Print(JSValueObject&& options, ReactPromise<JSValue> promise) noexcept
+    {
+      bool ok_to_go_ahead = true;
+      {
+        // Only support one print request at a time.
+        std::lock_guard<std::mutex> guard(printSync);
+        if (servingPrintRequest)
+        {
+          ok_to_go_ahead = false;
+        } else
+        {
+          servingPrintRequest = true;
+        }
+      }
+      if (!ok_to_go_ahead)
+      {
+        promise.Reject("Another print request is already being served.");
+        return;
+      }
+
+      RNPrintOptions printOptions;
+
+      if (options.find("html") != options.end())
+        printOptions.html = options["html"].AsString();
+
+      std::optional<std::string> filePath = std::nullopt;
+      if (options.find("filePath") != options.end())
+        printOptions.filePath = options["filePath"].AsString();
+
+      printOptions.isLandscape = (options.find("isLandscape") != options.end() ? options["isLandscape"].AsBoolean() : false);
+      printOptions.jobName = (options.find("jobName") != options.end() ? options["jobName"].AsString() : defaultJobName);
+
+      xaml::FrameworkElement root{ nullptr };
+
+      auto window = xaml::Window::Current();
```

Comment:
this means this module won't work from xaml islands (which might be an ok compromise).
Another way to do this would be to fall back to use XamlUIService.XamlRoot if Window::Current is null

Reply:
Will include the suggested workaround.
Is there a sample that uses xaml islands with react-native-windows that I could take a look into?

Reply:
@jaimecbernardo thanks! we don't have any modules that do so today since they mostly seem to work and not require access to something like Window::Current, but we do have code in the react-native-windows repo today that deals with this. See usages of XamlUIService.XamlRoot() and search for "isxamlisland" or something like that, we have a helper in the repo for it

Reply:
This is what I'm thinking of using :)
https://github.com/microsoft/react-native-windows/blob/df4cb47be1fcec7b5dc23ec032c0a75b19907d7f/vnext/Microsoft.ReactNative/Modules/LogBoxModule.cpp#L50-L52